### PR TITLE
[Auto-Techsupport] Minor Changes to Auto-Techsupport Documented

### PR DESCRIPTION
**Pull request by vivekrnv**
Changes include the following:

1) APP Extension Registry/De-Registry
2) Remove the stale information related to SONiC-SONIC upgrade
3) Section Related to Tech-support Locking mechanism

### CODE REVIEW

File: doc/auto_techsupport_and_coredump_mgmt.md [hunk 1]
```diff
@@ -462,96 +474,25 @@ Enhance the existing techsupport sonic-mgmt test with the following cases.
...
+Detailed Info related to Appliation Extension can be found here: https://github.com/Azure/SONiC/blob/master/doc/sonic-application-extension/sonic-application-extention-hld.md
+
+A new AUTO_TECHSUPPORT_FEATURE register/deregister option will be introduced. The existing FeatureRegistry class will be enahcned to also add/delete configuration related to AUTO_TECHSUPPORT_FEATURE table.
+
+This will be run when the application installs/uninstalls. Since, the auto-techsupport feature uses compile time flag to determine whether to enable/disable itself, it is not possible to determine that at runtime when the application is installed. 
+
+Thus the decision to whether or not to enable the new feature will be based on the current values of AUTO_TECHSUPPORT & AUTO_TECHSUPPORT_FEATURE tables. The default value for new feature will be disabled if the global state is shown disabled in init_cfg.json. If not, the feature will be enabled. The rate-limit-interval & memory threshold is set to 600 & 10% by default. 
+
+If there is a user-provided config in conifg_db.json or CONFIG_DB, it is not overwritten by the default config when the app installs and is not cleared when the app gets uninstalled.
```
Comment (stepanblyschak) [inline doc/auto_techsupport_and_coredump_mgmt.md]: Wonder what is the motivation? In the current design of sonic-package-manager the default behaviour is to delete all configuration related to the feature. User may choose to leave it if he uninstalls with "--keep-config". https://github.com/Azure/sonic-utilities/blob/master/sonic_package_manager/main.py#L417

Comment (vivekkarri) [reply doc/auto_techsupport_and_coredump_mgmt.md]: Only the default configuration is cleared when uninstall is invoked. If "--keep-config" is used (any user provided config + default config) stays. 

The idea is not to delete/overwrite any user provided config. 

Let's say user want to use different rate_limit_interval or available_mem_threshold than the default values, with this approach he can load the config anytime (i.e. before of after installation and can be sure that this is the configuration which is used)

Comment (stepanblyschak) [reply doc/auto_techsupport_and_coredump_mgmt.md]: All configuration is removed when keep_config is False, including user config, not just the default. This is by design.
https://github.com/Azure/sonic-utilities/blob/master/sonic_package_manager/service_creator/creator.py#L523

Comment (vivekkarri) [reply doc/auto_techsupport_and_coredump_mgmt.md]: Okay, i'll follow this then

Comment (vivekkarri) [reply doc/auto_techsupport_and_coredump_mgmt.md]: Handled. Please review

**Review by stepanblyschak — APPROVED** [review]

**Pull request closed without merging**